### PR TITLE
chore(connect-examples): manifest v3 web extensin with connect

### PR DESCRIPTION
**Pull request by karliatto**
## Description
This PR adds an example to `connect-examples` using Manifest V3 web extensions.

The main change is that since background scrips are not valid in Manifest V3 web extensions, in exchange they allow service workers and they do not have DOM so it is not possible to inject `iframe`. But service workers cannot access either webUSB neither trezor-bridge as it is now because of CORS. 
Support Trezor Connect in service worker would required to create a new transport or extend bridge in order to allow communication with device.

In contrast this PR creates a web extension where connect is loaded in an `.html` page from the extension, it is not loaded in the default `popup.html` from the extension because that one is closed when trezor popup is open therefore losing the context and not allowing getting response from trezor popup.

This has been only tested in Google Chrome browser.
In order to test it the web extension has to be loaded in Google Chrome browser and click on it to open new extension page that I called "Connect Manager". From that new page you can click a button to get features or get address and it should open a trezor popup window and get you back the correct payload.

## Related Issue

related https://github.com/trezor/trezor-suite/issues/6458 (maybe resolves, maybe not)

### CODE REVIEW

File: packages/connect-examples/webextension-v3/README.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# Web extension Manifest V3 example
+
+`@trezor/connect` running in an `html` page form the extension and communicating with `@trezor/popup` through `chrome.runtime` messages.
```
Comment (mroz22) [inline packages/connect-examples/webextension-v3/README.md]: `form` looks like a type

Comment (mroz22) [reply packages/connect-examples/webextension-v3/README.md]: I'd also add some more info here (similar to the other webextension readme) 
at least
```
## Install

-   `cd packages/connect-examples/webextension`
-   `yarn`
-   `node update.js`
```

and last but not least I would rename `connect-examples/webextension` to `connect-examples/webextension-mv2` and this newly added folder to `connect-examples/webextension-mv3`

File: packages/connect-examples/webextension-v3/manifest.json [hunk 2]
```diff
@@ -0,0 +1,14 @@
+{
+    "manifest_version": 3,
+    "name": "Trezor Suite",
```
Comment (mroz22) [inline packages/connect-examples/webextension-v3/manifest.json]: nitpick: I'd change this to something like
```
"name": "@trezor/connect webextension mv3 example",
```
is it too long? :thinking: 
and maybe unify it with mv2 example folder as well

File: packages/connect-examples/webextension-v3/style.css [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    background: #80808030;
+}
+li:nth-child(even) {
+    background: #ffffff;
+}
+
+h3,
+p {
+    margin: 0;
+}
```
Comment (mroz22) [inline packages/connect-examples/webextension-v3/style.css]: is it used? I haven't really noticed any styles :thinking:

File: packages/connect-examples/webextension-v3/trezor-usb-permissions.html [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+                position: relative;
+                width: 100%;
+                height: 100%;
+                min-height: 500px;
+                min-width: 328px;
+            }
+        </style>
+    </head>
+    <body>
+        <script type="text/javascript" src="./vendor/trezor-usb-permissions.js"></script>
```
Comment (mroz22) [inline packages/connect-examples/webextension-v3/trezor-usb-permissions.html]: nice, I just noticed that in mv2 folder this line looks like this: 

```
<script
            type="text/javascript"
            src="node_modules/@trezor/connect-web/lib/webextension/trezor-usb-permissions.js"
        ></script>
```

this is probably a mistake I made few days ago. if you are maybe changing stuff inside mv2 I'd dare to ask you to fix this too

File: packages/connect-examples/webextension-v3/update.js [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+});
+
+fetch('https://connect.trezor.io/9/trezor-connect.js')
+    .then(response => response.text())
+    .then(text => fs.writeFileSync(path.join(__dirname, 'vendor', 'trezor-connect.js'), text));
+
+const srcPath = path.join(__dirname, '../../connect-web/src/webextension');
+
+fs.copyFileSync(path.join(srcPath, 'trezor-content-script.js'), contentScriptPath);
+fs.copyFileSync(path.join(srcPath, 'trezor-usb-permissions.js'), usbPermissionsScriptPath);
```
Comment (mroz22) [inline packages/connect-examples/webextension-v3/update.js]: I thinking, maybe `packages/connect-examples/webextension-v3/trezor-usb-permissions.html` should be part of this utility script too? 

also, would it make sense to move this utility script one level up? and reuse it between examples?

**Review by mroz22 — CHANGES_REQUESTED** [review]
just a couple of nitpicks but it looks like a decent way to start with something

File: packages/connect-examples/webextension-v3/update.js (hunk 5 above)
Comment (karliatto) [reply packages/connect-examples/webextension-v3/update.js:31]: I think that makes sense, what about changing the file structure a bit like:

```
packages/connect-examples/webextension/mv2/
packages/connect-examples/webextension/mv3/
packages/connect-examples/webextension/update.js
```

What do you think?

Comment (karliatto) [reply packages/connect-examples/webextension-v3/update.js]: Yes, that makes sense, I added it to the update script bf8fe9d214b3b829e15c94252afe9535c66258b2

File: packages/connect-examples/webextension-v3/trezor-usb-permissions.html (hunk 4 above)
Comment (karliatto) [reply packages/connect-examples/webextension-v3/trezor-usb-permissions.html]: I added it 30acbf062b5d9fd85af5f8825bf859e654cd5ecb along with other changes to make both extension examples following same pattern.

File: packages/connect-examples/webextension-v3/style.css (hunk 3 above)
Comment (karliatto) [reply packages/connect-examples/webextension-v3/style.css]: Removed that https://github.com/trezor/trezor-suite/pull/7699/commits/420dd87e3e07271febe4be829f7b3ecfbf313c37

File: packages/connect-examples/webextension-v3/manifest.json (hunk 2 above)
Comment (karliatto) [reply packages/connect-examples/webextension-v3/manifest.json]: Updated https://github.com/trezor/trezor-suite/pull/7699/commits/420dd87e3e07271febe4be829f7b3ecfbf313c37

File: packages/connect-examples/webextension-v3/README.md (hunk 1 above)
Comment (karliatto) [reply packages/connect-examples/webextension-v3/README.md]: Included in README https://github.com/trezor/trezor-suite/pull/7699/commits/420dd87e3e07271febe4be829f7b3ecfbf313c37

**Review by mroz22 — APPROVED** [review]

**Pull request merged**